types/recommendation.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CStorClusterRecommendation is a kubernetes custom
+// resource that represents recommended configurations
+// for one given input/requirement.
+type CStorClusterRecommendation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   CStorClusterRecommendationSpec   `json:"spec"`
```

Comment:
Should we put in the CStorClusterRecommendationRequestSpec here? This will help us to know for what request, this recommendation was created.

Reply:
Ok. Perhaps `RequestReference` that in turn can have Namespace, Name, UID, etc

Reply:
cc @shovanmaity

Reply:
For this phase, we will not have this as a cr for that we need to have one alternative way.
Also, @AmitKumarDas does CCC take group details as input?

I want to provision one raidz pool with 9 device count is 3 so what will be the combination?
 **3 * ( 2 + 1 )** or **8 + 1**
How CCC will work for this condition?

Reply:
will it create one group of 8+1 or 3 groups of 3+1?

Reply:
@avishnu gave this point to me in one offline discussion.